cpu/cortexm_common/ldscripts/cortexm.ld

```diff
@@ -14,16 +15,20 @@
  * @brief           Memory definitions for the Cortex-M family
  *
  * @author          Francisco Acosta <francisco.acosta@inria.fr>
+ *                  Gaëtan Harter <gaetan.harter@inria.fr>
  *
  * @}
  */
 
-_boot_offset = DEFINED( _rom_offset ) ? _rom_offset : 0x0 ;
+_rom_offset = DEFINED( _rom_offset ) ? _rom_offset : 0x0 ;
+_fw_rom_length = DEFINED( _fw_rom_length ) ? _fw_rom_length : _rom_length - _rom_offset;
+
+ASSERT((_fw_rom_length <= _rom_length - _rom_offset), "Specified _fw_rom_length does not fit in ROM");
```

Comment:
Maybe reword the assert message to something like
`Specified firmware size does not fit in ROM`
to keep it understandable for end users.